server/routerlicious/packages/lambdas/src/alfred/index.ts

```diff
@@ -1053,7 +1072,10 @@ export function configureWebSocketServices(
 		// Message sent when a new signal is submitted to the router
 		socket.on(
 			"submitSignal",
-			(clientId: string, contentBatches: (IDocumentMessage | IDocumentMessage[])[]) => {
+			(
+				clientId: string,
+				contentBatches: (IDocumentMessage | IDocumentMessage[])[] | ISentSignalMessage[],
```

Comment:
When does and form of `IDocumentMessage` ever get here?

Reply:
Current (soon to be old) drivers still send signals of type IDocumentMessage, so I'm keeping this for backwards compatibility with these v1 signals.

Reply:
Do they actually, or was that just the type stamped on top of whatever they actually send?

Reply:
We can't really trust types set on things that are over the wire. I think a lot of people are a bit loose with use of `as`. And I think the two types are similar and probably got mixed up.
```typescript
export interface IDocumentMessage {
	// Required fields
	clientSequenceNumber: number;
	referenceSequenceNumber: number;
	type: string;
	contents: unknown;
	// Optional
	metadata?: unknown;
	serverMetadata?: unknown;
	traces?: ITrace[];
	compression?: string;
}
```
and pre-update
```typescript
export interface ISignalMessageBase {
	clientConnectionNumber?: number;
	referenceSequenceNumber?: number;
	type?: string;
	content: unknown;
}
```

Reply:
The type is really ISignalEnvelope from the container runtime, which looks to be a stripped-down version of IDocumentMessage.

Reply:
and ISignalMessage wraps an ISignalEnvelope in it's `content` attribute. [`ISignalMessage.content.contents.content` is the message that the client eventually receives ](https://github.com/microsoft/FluidFramework/blob/bf7ae34b81cb28099a765311423071bbe697a830/packages/runtime/container-runtime/src/containerRuntime.ts#L2374C3-L2378C5)

Reply:
```typescript 
public processSignal(message: ISignalMessage, local: boolean) {
    const envelope = message.content as ISignalEnvelope;
    const transformed: IInboundSignalMessage = {
	    clientId: message.clientId,
	    content: envelope.contents.content,
	    type: envelope.contents.type,
    };

...
```

Reply:
Do you know anything related to this @ChumpChief ?

Reply:
Not sure, but the typing does look pretty questionable (even just comparing typing of `IDocumentDeltaConnection.submitSignal` (any) vs. `DocumentDeltaConnection.submitSignal` (IDocumentMessage)).  Probably want to talk to someone with more server experience to understand the server's current expectations and norm everything to that as a starting point.

This is a good place to exercise caution, since whatever the payload is, it's fresh off the wire.